task2

Nordennavic opened the pull request.

Nordennavic committed:
Попытался сделать 2 лабораторную.

Nordennavic committed:
Забрыл убрать //code из второй функции.

AlexNg9 commented on index.js:
```diff
@@ -6,7 +6,11 @@
  */
 function customBind (func, context, ...args) {
-    // code
+    const primaryArgs = [].concat(args);
```
args – уже массив. Зачем?

AlexNg9 commented on index.js:
```diff
@@ -6,7 +6,11 @@
+    const primaryArgs = [].concat(args);
+
+    return function (...newArgs) {
+        return func.apply(context, primaryArgs.concat([].concat(newArgs)));
```
Зачем [].concat? Почему на присоединить сразу к primary?

AlexNg9 commented on index.js:
```diff
@@ -19,7 +23,18 @@ function customBind (func, context, ...args) {
  */
 function sum (x) {
-    // code
+    if (typeof x === 'undefined') { return 0; }
```
Зачем через typeof?

AlexNg9 replied:
Лучше без фигурных, если в одну строку, либо на новую строку.

AlexNg9 commented on index.js:
```diff
@@ -19,7 +23,18 @@ function customBind (func, context, ...args) {
+    if (typeof x === 'undefined') { return 0; }
+
+    let tempSum = x;
+    function callBack (newTemp) {
```
callback – одно слово.
Лучше тогда newTempSum.

Nordennavic committed:
customBind:
-Убрал лишние приведения к массиву через []
sum:
-Стилистически изменил однострочный if
-Оптимизировал условие в if'aх
-Переименовал функцию

AlexNg9 commented on index.js:
```diff
@@ -19,7 +21,20 @@ function customBind (func, context, ...args) {
+        return 0;
+    }
+
+    let tempSum = x;
```
temptemptemp – плохие названия.

AlexNg9 replied:
Попробуйте без этой переменной переписать.

AlexNg9 commented on index.js:
```diff
@@ -19,7 +21,20 @@ function customBind (func, context, ...args) {
+            tempSum += newTemp;
+
+            return newTempSum;
+        } else return tempSum;
```
Лучше вернуть tempSum, если newTemp === undefined, а ниже остальную логику. Так лучше читаемость.

Nordennavic committed:
Update:
1)Переписал без temp
2)Поменял местами логику в if'e
3)Переименовал функцию и переменную

AlexNg9 commented on index.js:
```diff
@@ -25,16 +25,17 @@ function sum (x) {
+    function updateSum (addend) {
+        if (addend === undefined) {
+            return x;
+        } else {
```
else избыточен
